spring-boot-tests/spring-boot-smoke-tests/spring-boot-smoke-test-pulsar/src/test/java/smoketest/pulsar/SamplePulsarApplicationPemBasedSslTests.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2012-2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package smoketest.pulsar;
+
+import java.time.Duration;
+import java.util.stream.IntStream;
+
+import org.awaitility.Awaitility;
+import org.junit.jupiter.api.Test;
+import org.testcontainers.junit.jupiter.Container;
+import org.testcontainers.junit.jupiter.Testcontainers;
+import smoketest.pulsar.SamplePulsarApplication.SampleMessageConsumer;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.DynamicPropertyRegistry;
+import org.springframework.test.context.DynamicPropertySource;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+@SpringBootTest(
+		properties = { "spring.pulsar.client.ssl.bundle=client", "spring.pulsar.administration.ssl.bundle=client",
```

Comment:
I have added an SSL smoke test pair (JKS+PEM) for both imperative and reactive. 

I see that for data cassandra we always have reactive+ssl  and imperative+ssl smoke test - but none w/o SSL. I think having one w/o SSL is valuable though as not everyone will be using SSL.

Wdyt?

Reply:
We've decided to leave SSL out for now, although we'll document how to set it up using a customizer.